cmd/virt-handler/virt-handler.go

```diff
@@ -176,6 +179,27 @@ func (app *virtHandlerApp) markNodeAsUnschedulable(logger *log.FilteredLogger) {
 	}
 }
 
+// Using the downward API, look for dedicated migration network migration0 and, if found, set migration IP to its IP
+func findMigrationIP(migrationIp *string) error {
+	var networkStatus []virthandler.NetworkStatus
+
+	dat, err := ioutil.ReadFile("/etc/podinfo/network-status")
+	if err != nil {
+		return fmt.Errorf("failed to read network status from downwards API")
+	}
+	err = yaml.Unmarshal(dat, &networkStatus)
```

Comment:
Do we rely on some implicit behaviour that we can deal with empty files? Because it looks like we would crate this file always and in case that there is no annotation the file is presumably empty?

Reply:
The file will always contain at least 1 entry for the main network.
Also yes, `yaml.Unmarshall()` will happily decode an empty string into an empty list.

Reply:
Even if multus is not installed? The annotation looks very multus specific.

Reply:
The annotation is multus-specific, but the network status isn't.
If the annotation is not present and/or multus is not installed, the network-status file will still show the status of the main network.
Or maybe I misunderstood your question? :)

Reply:
Well maybe I misunderstand something, but 

```
	container.VolumeMounts = append(container.VolumeMounts, corev1.VolumeMount{
		Name:      "podinfo",
		MountPath: "/etc/podinfo",
	})
	pod.Volumes = append(pod.Volumes, corev1.Volume{
		Name: "podinfo",
		VolumeSource: corev1.VolumeSource{
			DownwardAPI: &corev1.DownwardAPIVolumeSource{
				Items: []corev1.DownwardAPIVolumeFile{
					{
						Path: "network-status",
						FieldRef: &corev1.ObjectFieldSelector{
							FieldPath: `metadata.annotations['k8s.v1.cni.cncf.io/network-status']`,
						},
					},
				},
			},
		},
	})
	```
	
seems to be the source of the `network-status` file and that one looks to me like it would only contain any context if this annotation exists. And this annotation does probably not exist if there is not multus installed?

Reply:
Sorry, you are absolutely right!
Without multus, /etc/podinfo/network-status still gets created but is empty.
As I said before though, this case is properly handled by the code, though not explicitly.

Reply:
Added unit tests

Reply:
Thanks! :)